src/jdbc/cubrid/jdbc/jci/UStatement.java

```diff
@@ -811,19 +813,39 @@ private void readResultMeta(UInputBuffer inBuffer) throws UJciException {
 		}
 	}
 
-	private void fetchResultData(UInputBuffer inBuffer) throws UJciException {
+	private void fetchResultData(UInputBuffer inBuffer, UStatementCacheData cacheData) throws UJciException {
 		executeResult = inBuffer.getResCode();
 		if (maxFetchSize > 0) {
 			executeResult = Math.min(maxFetchSize, executeResult);
 		}
 		totalTupleNumber = executeResult;
 		batchParameter = null;
 
+		//jdbc cache feature
+		if (cacheData != null && resultInfo.length == 1) {
+			stmt_cache_data = cacheData;
+			stmt_cache_data.setCacheData(totalTupleNumber, resultInfo);
+		}
+		else if (resultInfo.length > 1) {
+			result_cacheable = false;
+		}
+		//--
+		
 		if (commandTypeIs == CUBRIDCommandType.CUBRID_STMT_SELECT
 		        && totalTupleNumber > 0) {
 			inBuffer.readInt(); // fetch_rescode
 			read_fetch_data(inBuffer, UFunctionCode.FETCH);
 		}
+		
+		//jdbc cache feature
+		if (stmt_cache_data != null) {
+			int total_size = fetchedSize + relatedConnection.getUrlCache().getCacheSize();
+			if (total_size < relatedConnection.getUrlCache().getLimit()) {
+				stmt_cache_data.setCacheData(tuples, currentFirstCursor, fetchedTupleNumber, fetchedSize);
```

Comment:
`addCacheData` would be better for readability

Reply:
Ok